javascript/atoms/dom.js

```diff
@@ -574,24 +574,26 @@ bot.dom.isShown = function(elem, opt_ignoreOpacity) {
       if (bot.dom.getEffectiveStyle(e, 'display') == 'none') {
         return false;
       }
-      var parent;
-      do {
-        parent = bot.dom.getParentNodeInComposedDom(e);
-        if (parent instanceof ShadowRoot) {
-          if (parent.host.shadowRoot != parent) {
-            // There is a younger shadow root, which will take precedence over
-            // the shadow this element is in, thus this element won't be
-            // displayed.
-            return false;
-          } else {
-            parent = parent.host;
-          }
-        } else if (parent && (parent.nodeType == goog.dom.NodeType.DOCUMENT ||
-            parent.nodeType == goog.dom.NodeType.DOCUMENT_FRAGMENT)) {
-          parent = null;
+
+      var parent = bot.dom.getParentNodeInComposedDom(e);
+
+      if (parent instanceof ShadowRoot) {
+        if (parent.host.shadowRoot != parent) {
+          // There is a younger shadow root, which will take precedence over
+          // the shadow this element is in, thus this element won't be
+          // displayed.
+          return false;
+        } else {
+          parent = parent.host;
         }
-      } while (elem && elem.nodeType != goog.dom.NodeType.ELEMENT);
```

Comment:
i removed this loop because `elem` was never modified, so it would either run once or forever... guess we never ended up with this being as truthy

Reply:
we have some recursion below to deal with traversing the tree anyway, so i guess this removal is fine.